pkg/controller/instanceset/instance_util.go

```diff
@@ -263,7 +317,60 @@ func GenerateInstanceNames(parentName, templateName string,
 			}
 		}
 	}
-	return instanceNameList, ordinal
+	return instanceNameList
+}
+
+func GetIndexPointsByTemplateName(templatesIndexRanges []workloads.InstanceTemplateIndexRanges, name string) ([]int32, error) {
+	if len(templatesIndexRanges) == 0 {
+		return nil, nil
+	}
+	indexRanges, err := GetTemplateIndexRangesByTemplateName(templatesIndexRanges, name)
+	if err != nil {
+		return nil, err
+	}
+	return ConvertIndexRangesToIndexPoints(indexRanges)
+}
+
+func GetTemplateIndexRangesByTemplateName(templatesIndexRanges []workloads.InstanceTemplateIndexRanges, name string) ([]string, error) {
+	for _, templateIndexRanges := range templatesIndexRanges {
+		if templateIndexRanges.Name == name {
+			return templateIndexRanges.IndexRanges, nil
+		}
+	}
+	return nil, fmt.Errorf("template %s not found", name)
+}
+
+func ConvertIndexRangesToIndexPoints(indexRanges []string) ([]int32, error) {
+	indexPoints := sets.Set[int32]{}
+	for _, segment := range indexRanges {
+		segmentNumbers := strings.Split(segment, "-")
+		if len(segmentNumbers) != 2 {
+			return nil, fmt.Errorf("indexRanges must have only one - character, but now have %v", len(segmentNumbers))
```

Comment:
it's better to support format like this: ["1-3", "5"]

Reply:
It is recommended to define the structure explicitly rather than using formatted strings:
```go
type Range struct {
    Start int32
    End int32
}

type Ordinals struct {
    Ranges []Range
    Discrete []int32
}

Reply:
@leon-inf I don’t understand the purpose of `Discrete` being placed here. Could you explain it in detail?

Reply:
Using the ```Discrete``` field to represent discrete ordinals explicitly, such as "5" mentioned in the example above, rather than using a range with the same start and end to represent it.